mpl_draw requires all keyword arguments to be specified to pass mypy

### Information

- **rustworkx version**: 0.14.2
- **Python version**:  3.9.12
- **Rust version**: N/A
- **Operating system**: Windows 10 build 19045

### What is the current behavior?
When using `mypy` v1.8.0 with rustworkx it complains about missing keyword arguments with a call to `mpl_draw`:

```
error: Missing named argument "arrowstyle" for "mpl_draw"  [call-arg]
error: Missing named argument "arrow_size" for "mpl_draw"  [call-arg]
error: Missing named argument "node_list" for "mpl_draw"  [call-arg]
error: Missing named argument "edge_list" for "mpl_draw"  [call-arg]
error: Missing named argument "node_size" for "mpl_draw"  [call-arg]
error: Missing named argument "node_color" for "mpl_draw"  [call-arg]
error: Missing named argument "node_shape" for "mpl_draw"  [call-arg]
error: Missing named argument "alpha" for "mpl_draw"  [call-arg]
error: Missing named argument "cmap" for "mpl_draw"  [call-arg]
error: Missing named argument "vmin" for "mpl_draw"  [call-arg]
error: Missing named argument "vmax" for "mpl_draw"  [call-arg]
error: Missing named argument "linewidths" for "mpl_draw"  [call-arg]
error: Missing named argument "edge_color" for "mpl_draw"  [call-arg]
error: Missing named argument "edge_cmap" for "mpl_draw"  [call-arg]
error: Missing named argument "edge_vmin" for "mpl_draw"  [call-arg]
error: Missing named argument "edge_vmax" for "mpl_draw"  [call-arg]
error: Missing named argument "style" for "mpl_draw"  [call-arg]
error: Missing named argument "font_size" for "mpl_draw"  [call-arg]
error: Missing named argument "font_color" for "mpl_draw"  [call-arg]
error: Missing named argument "font_weight" for "mpl_draw"  [call-arg]
error: Missing named argument "font_family" for "mpl_draw"  [call-arg]
error: Missing named argument "label" for "mpl_draw"  [call-arg]
error: Missing named argument "connectionstyle" for "mpl_draw"  [call-arg]
```

### What is the expected behavior?
It shouldn't be expected for one to list all _optional_ keyword arguments to `mpl_draw`.

### Steps to reproduce the problem
Run `mypy` on some code that calls `mpl_draw`.

--- a/releasenotes/notes/fix-mpl-draw-kargs-typing-6cf7cfde0db072d5.yaml
+++ b/releasenotes/notes/fix-mpl-draw-kargs-typing-6cf7cfde0db072d5.yaml
@@ -1,0 +1,6 @@
+---
+fixes:
+  - |
+    Fixed a bug in the type hint for the :func:`~rustworkx.visualization.mpl_draw`.
+    Previously, all `kwargs` were required when calling the method. The new annotation
+    accepts `kwargs` with partial arguments.

--- a/rustworkx/visualization/matplotlib.pyi
+++ b/rustworkx/visualization/matplotlib.pyi
@@ -20,7 +20,7 @@ if typing.TYPE_CHECKING:
 _S = typing.TypeVar("_S")
 _T = typing.TypeVar("_T")
 
-class _DrawKwargs(typing.TypedDict):
+class _DrawKwargs(typing.TypedDict, total=False):
     arrowstyle: str
     arrow_size: int
     node_list: list[int]

